backend/app/main.py

```diff
@@ -13,25 +16,54 @@
 from pyjwt_key_fetcher import AsyncKeyFetcher
 from starlette.middleware.sessions import SessionMiddleware
 
-from .settings import conf
+key_fetcher = AsyncKeyFetcher()
+oauth = OAuth()
+
+
+async def register_oauth():
+    dataspace_configuration = await get_dataspace_configuration()
+    authentication_providers = dataspace_configuration["authentication_providers"]
+    authentication_providers_end_user = authentication_providers["end_user"]
+    authentication_providers_end_user_base_url = authentication_providers_end_user[
+        "base_url"
+    ]
```

Comment:
I guess I'd here as well would have extracted the URL at once, something like 
```python
    oidc_provider_url = dataspace_configuration["authentication_providers"]["end_user"][
        "base_url"
    ]
```
But this also works like it is, so you can just mark it as resolved with no change.